src/components/atoms/TextInput.vue

```diff
@@ -41,21 +55,45 @@ export default {
   emits: ["add-message"],
   setup(_, context) {
     const text = ref("");
+    const iconState = ref("SendInactive");
+
+    //these refs are reassigned to their respective components on mounting
+    //as with other refs, the actual elements are accessed with .value (input.value, for example)
+    const input = ref(null);
+    const sendMsgBtn = ref(null);
```

Comment:
Instead of creating a ref to the button we manipulate for tab index we can have a variable that we set. IE var sendMessageTabIndex. Then :index="sendMessageTabIndex"

Reply:
Right, that makes more sense.